File: Team5/Akanksha_Singh/Week-1/LongestSubstring.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       int arr[256]={0};
       int i=0, j=0, count=0; 
        while(j<s.length()){
            if(!arr[s[j]]){
                arr[s[j++]]++;
                count=max(j-i, count);
            }
            else{
                arr[s[i]]--;
                i++;
            }
        }
        return count;
    }
};

File: Team2/kusumlata/longestsubstring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int a[] = new int[256];
        int i=0;
        int j=0;
        int max = 0;
        while(i<n){
            if(a[s.charAt(j)] == 0){
                a[s.charAt(j)]++;
                j++;
                max = Math.max(max , j-i);
                if(j == n)
                    break;
            }
            else{
                a[s.charAt(i)]--;
                i++;
            }
        }
        return max;
    
        
    }
}
